src/transformStateWithClones.js

```diff
@@ -8,6 +8,29 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const stateHistory = [];
+
+  let currentState = { ...state };
```

Comment:
The naming for the object copy can be improved for clarity. Instead of `currentState`, a more descriptive name like `stateCopy` would better convey the purpose of this variable.